### ISSUE TITLE
BatchDataset cannot handle arguments from tuple

### ISSUE THREAD

**Issue by escorciav** [pytorch/tnt#16]
Hi,

I tried to use [BatchDataset](https://github.com/pytorch/tnt/blob/master/torchnet/dataset/batchdataset.py#L6) to concatenate the outputs, `tuple` with features, labels, etc., from my dataset but it seems that I should pack them inside a dictionary otherwise, [this line](https://github.com/pytorch/tnt/blob/master/torchnet/transform.py#L24-L25) will produce an error. 
It would be cool to say that in the documentation or provide support for iterable like tuple and list.

P.D. Thanks for providing this awesome set of tools.

**Comment by RicherMans** [comment]
As a small code example:
```python
import torchnet as tnt
import torch

f = []
t = []
for i in xrange(100):
	f.append(torch.randn(20))
	t.append(torch.randn(1))
b = tnt.dataset.TensorDataset([f,t])
seta = tnt.dataset.BatchDataset(b, 2)
for k,v in seta:
	print k,v
```
This will fail in the line above mentioned.
